File: crates/cli/tests/fixtures/rule_context/scala/Controller.scala
import javax.naming.ldap.Rdn
import javax.naming.directory.DirContext
import javax.naming.directory.SearchControls
import javax.servlet.http.HttpServletRequest
class Controller {
  def direct(request: HttpServletRequest, directory: DirContext): Unit = {
    val name = request.getParameter("name")
    directory.search("dc=example", "(uid=" + name + ")", new SearchControls())
  }
  def escaped(request: HttpServletRequest, directory: DirContext): Unit = {
    val name = request.getParameter("name")
    directory.search("dc=example", "(uid=" + Rdn.escapeValue(name) + ")", new SearchControls())
  }
}

File: crates/cli/tests/fixtures/rule_context/java/Controller.java
import org.springframework.web.bind.annotation.RequestParam;
import javax.naming.ldap.Rdn;
import javax.naming.directory.DirContext;
import javax.naming.directory.SearchControls;
class Controller {
    void escaped(@RequestParam String name, DirContext directory) throws Exception {
        directory.search("dc=example", "(uid=" + Rdn.escapeValue(name) + ")", new SearchControls());
    }
    void direct(@RequestParam String name, DirContext directory) throws Exception {
        directory.search("dc=example", "(uid=" + name + ")", new SearchControls());
    }
}
